### PR TITLE
setTimeout and setInterval on window should return numeric ID

### PR DESCRIPTION
**Pull request by alvarorahul**
initial draft, still trying to figure out the tests

### CODE REVIEW

File: lib/jsdom/browser/Window.js [hunk 1]
```diff
@@ -402,29 +402,26 @@ function matchesDontThrow(el, selector) {
   }
 }
 
+let latestTimerId = 0;
```
Comment (Sebmaster) [inline lib/jsdom/browser/Window.js]: This will make all instances have a shared timer-id. Not sure we want that.

**Comment by alvarorahul** [comment]
@Sebmaster thanks for your feedback, I've sent out a new iteration

File: test/web-platform-tests/to-upstream/html/webappapis/timers/settimeout-setinterval-handles.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  assert_equals(c, b + 1);
+
+  const d = setInterval(() => {}, 1000);
+  assert_equals(d, c + 1);
+
+  clearTimeout(a);
+  clearTimeout(b);
+  clearInterval(c);
+  clearInterval(d);
+});
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/html/webappapis/timers/settimeout-setinterval-handles.html]: Test needs a name

File: test/web-platform-tests/to-upstream/html/webappapis/timers/settimeout-setinterval-handles.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+<link rel="author" title="Alvaro Dias" href="mailto:alvarorahul@gmail.com">
+<link rel="help" href="https://html.spec.whatwg.org/multipage/#dom-windowtimers-settimeout">
+<link rel="help" href="https://heycam.github.io/webidl/#es-operations">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<script>
+"use strict";
+
+test(() => {
```
Comment (domenic) [inline test/web-platform-tests/to-upstream/html/webappapis/timers/settimeout-setinterval-handles.html]: Should use async_test or promise_test and make sure all the non-cleared callbacks are actually executed (and that none of the cleared ones are executed).

File: lib/jsdom/browser/Window.js [hunk 4]
```diff
@@ -50,7 +50,7 @@ function Window(options) {
   // vm initialization is defered until script processing is activated (in level1/core)
   this._globalProxy = this;
 
-  this.__timers = [];
+  this.__timers = {};
```
Comment (domenic) [inline lib/jsdom/browser/Window.js]: Nit: use Object.create(null) for these sorts of things, here and when doing stopAllTimers.

**Comment by domenic** [comment]
This is looking pretty good, just a tiny bit more test coverage would be lovely :)

File: lib/jsdom/browser/Window.js (hunk 4 above)
Comment (alvarorahul) [inline lib/jsdom/browser/Window.js]: addressed review comments

**Comment by domenic** [comment]
LGTM, awesome. Will merge tonight-ish, certainly in time for the next release.

**Comment by domenic** [comment]
Merged as f53876aa0368b3debb73272db5023708e53b24fb, thank you! BTW you may want to add your Git email to GitHub so you get credit for the commits: https://help.github.com/articles/adding-an-email-address-to-your-github-account/

**Pull request closed without merging**